spec/controllers/v0/user_action_events_controller_spec.rb

```diff
@@ -0,0 +1,201 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe V0::UserActionEventsController, type: :controller do
+  include RequestHelper
+
+  context 'when not logged in' do
+    it 'returns unauthorized' do
+      get :index
+      expect(response).to have_http_status(:unauthorized)
+    end
+  end
+
+  describe 'GET #index' do
+    let(:idme_uuid) { 'some-idme-uuid' }
+    let(:user) { create(:user, idme_uuid:) }
+    let!(:user_verification) { create(:idme_user_verification, idme_uuid:) }
+
+    before do
+      sign_in_as(user)
+    end
+
+    context 'when there are user actions' do
+      let(:user_action_event_one) { create(:user_action_event, details: 'Sample event 1') }
+      let(:user_action_event_two) { create(:user_action_event, details: 'Sample event 2') }
+
+      let!(:user_action_one) do
+        create(:user_action, subject_user_verification_id: user_verification.id,
+                             user_action_event: user_action_event_one, created_at: 1.year.ago)
+      end
+      let!(:user_action_two) do
+        create(:user_action, subject_user_verification_id: user_verification.id,
+                             user_action_event: user_action_event_two, created_at: 2.months.ago)
+      end
+      let!(:user_action_three) do
+        create(:user_action, subject_user_verification_id: user_verification.id,
+                             user_action_event: user_action_event_one, created_at: 3.weeks.ago)
+      end
+      let!(:user_action_four) do
+        create(:user_action, subject_user_verification_id: user_verification.id,
+                             user_action_event: user_action_event_two, created_at: 4.days.ago)
+      end
+
+      let(:page) { 1 }
```

Comment:
You can remove `page` & `per_page` and define them only when you need to for the pagination tests.